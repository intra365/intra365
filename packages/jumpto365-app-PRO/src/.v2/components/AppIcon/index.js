import React, { Component } from 'react';
import PropTypes from 'prop-types'
import AppIconGeneric from '../AppIconGeneric'
import Jumpto365Service from '../../services/index'
import Jumpto365ServiceApp from '../_Contexts/Jumpto365App'
export default class AppIcon extends Component {
    static propTypes  = {
        name :   PropTypes.string.isRequired,
        size : PropTypes.number


    }

constructor(props) {
    super(props);
    this.state = {item : {},loaded:false}
}


    _onJumpTo = () => {
            console.log(this.state.item.Link)

    }
    _onClick = () => {
        console.log(this.state.item.name)


    }

    componentDidMount = () => {
        var jumpto365Service = new Jumpto365Service()
        var item = jumpto365Service.getApp(this.props.name)
       
            this.setState({item,loaded: true})
       
       //     Jumpto365ServiceApp.emitError(this,err,"AppIconLoader")

       

    }
render (){
    if (!this.state.loaded) return "NOT LOADED"
    var s = this.state.item

    //return JSON.stringify(s)
    return (
    <AppIconGeneric 
                    size={this.props.size} 
                    backgroundColor={s.color} 
                    name="word" 
                    title={s.name}
                    description="Word processing in the browser."  
                    iconUrl={s.icon} />)

    // <AppIconGeneric size={this.props.size} 
    //                 name={this.state.item.name} 
    //                 backgroundColor={s.color}
    //                 iconUrl = {s.icon}
    //                 title={s.name}



    //                 onJumpTo={this._onJumpTo} 
    //                 onClick={this._onClick}/>
    
}

}
