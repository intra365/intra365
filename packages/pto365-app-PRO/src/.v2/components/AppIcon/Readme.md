



```js
<AppIcon size="128" name="word" />
```


```js
<AppIcon size="200" name="excel" />
```


```js
<AppIcon size="200" name="exceler" />
```
