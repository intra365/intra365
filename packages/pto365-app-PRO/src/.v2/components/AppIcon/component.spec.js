import React, { Component } from 'react';

import AppIcon from '.'


test('Check that it renders 320 px', () => {

    const wrapper = shallow(<AppIcon name="word" size={320} />);

    expect(wrapper).toMatchSnapshot();
 
}

)  



