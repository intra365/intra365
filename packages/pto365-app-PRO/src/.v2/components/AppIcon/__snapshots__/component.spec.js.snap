// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 320 px 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppIcon
    name="word"
    size={320}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColor": "#2b579a",
      "description": "Word processing in the browser.",
      "iconUrl": "https://jumpto365.com/resources/images/app/Word.png",
      "name": "word",
      "size": 320,
      "title": "word",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColor": "#2b579a",
        "description": "Word processing in the browser.",
        "iconUrl": "https://jumpto365.com/resources/images/app/Word.png",
        "name": "word",
        "size": 320,
        "title": "word",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
